lib/providers/_utils.js

```diff
@@ -21,8 +21,8 @@ function addAuthorize (strategy) {
   // Set response_type to code
   strategy.response_type = 'code'
 
-  // Json parser
-  const jsonMiddleware = bodyParser.json()
+  // Form data parser
+  const formMiddleware = bodyParser.urlencoded()
```

Comment:
This line is affecting all providers! If formMiddleware is required we can add an optional parameter like `addAuthorize(strategy, { json: true } = {})` and conditionally use either form for json middleware

Reply:
Currently, the `addAuthorize` function is only called by the github provider. The problem is the `_handleCallback` method in the oauth2 scheme. It looks like it [used to send json data](https://github.com/nuxt-community/auth-module/blob/1a8a92418aba04c97ae75830add54cc7f69987e4/lib/auth/schemes/oauth2.js#L99-L101) but at some point was [changed to send form data](https://github.com/nuxt-community/auth-module/blob/dev/lib/schemes/oauth2.js#L103-L109). Whenever this was changed, the `serverMiddleware` was not updated to use the form middleware. Currently, I could not get the github provider to work without this change.

As a side not, this was an issue I ran into with the `laravel.passport` provider as well and was a major reason I did not use the `addAuthorize` function. If we also had the authorization request send the full `req.body` instead of just `req.body.code` it could potentially be reused for the `laravel.passport` provider.